addons/binding/org.openhab.binding.co2signal/src/main/java/org/openhab/binding/co2signal/handler/CO2SignalHandler.java

```diff
@@ -0,0 +1,292 @@
+
+            if (location != null && !"".equals(location)) {
+                location = location.replace(" ", "").replace("\"", "").replace("'", "").trim();
+                String[] latLon = location.split(",");
+                urlStr = urlStr.replace("%QUERY%", "lat=" + latLon[0] + "&lon=" + latLon[1]);
+            } else {
+                urlStr = urlStr.replace("%QUERY%", "countryCode=" + countryCode);
+            }
+
+            logger.debug("URL = {}", urlStr);
+
+            // Run the HTTP request and get the JSON response from co2cn.org
+            URL url = new URL(urlStr);
+            HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
+            connection.setRequestProperty("User-Agent", "Mozilla/5.0");
+            connection.setRequestProperty("auth-token", config.apikey);
+
+            try {
+                String response = IOUtils.toString(connection.getInputStream());
+                logger.debug("co2Response = {}", response);
+
+                // Map the JSON response to an object
+                result = gson.fromJson(response, CO2SignalJsonResponse.class);
+            } finally {
+                IOUtils.closeQuietly(connection.getInputStream());
+            }
+
+            if (result == null) {
+                errorMsg = "no data returned";
+            } else if (result.getData() != null && result.getStatus() != "error") {
+                resultOk = true;
+            } else {
+                errorMsg = "missing data sub-object";
+            }
+
+            if (!resultOk) {
+                logger.warn("Error in co2cn.org (CO2 Signal) response: {}", errorMsg);
+            }
+        } catch (MalformedURLException e) {
+            errorMsg = e.getMessage();
+            logger.warn("Constructed url is not valid: {}", errorMsg);
+        } catch (JsonSyntaxException e) {
+            errorMsg = "Configuration is incorrect";
+            logger.warn("Error running co2cn.org (CO2 Signal) request: {}", errorMsg);
+        } catch (IOException | IllegalStateException e) {
+            errorMsg = e.getMessage();
+        }
+
+        // Update the thing status
+        if (resultOk) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, "OK");
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, errorMsg);
+        }
+
+        return resultOk ? result : null;
+    }
+
+    public static Object getValue(String channelId, CO2SignalJsonResponse data) throws Exception {
+        String[] fields = StringUtils.split(channelId, "#");
+
+        if (data == null) {
+            return null;
+        }
+
+        String fieldName = fields[0];
+
+        switch (fieldName) {
+            case CO2SignalBindingConstants.COUNTRYCODE:
+                return data.getCountryCode();
+            case CO2SignalBindingConstants.CARBONINTENSITY:
+                return data.getData().getCarbonIntensity();
```

Comment:
Could data.getData() be null? In that case, you would run into a NullPointerException here and in the next case.

Reply:
Fixed